libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3409,16 +3410,22 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 
 	return m_functionCollector.createFunction(functionName, [&]() {
 		Whiskers templ(R"(
-			function <functionName>(value<?fromCalldataDynamic>, length</fromCalldataDynamic>) -> converted {
+			function <functionName>(value<?fromCalldataDynamic>, length</fromCalldataDynamic>) -> converted <?toCalldataDynamic>, outLength</toCalldataDynamic> {
 				<body>
+				<?toCalldataDynamic>
```

Comment:
This doesn't look like a no-op - was there another bug?

Reply:
There was assertion in this function that  `_to` is not in calldata and was failing with that.
And it is actually no-op as in case of calldata to calldata it just returns both (or single) value(s) from arguments...

Reply:
Ah ok, now I understand it. Can you add more assertions such that the only conversion is between byte arrays or between arrays of equal type?

Reply:
I added assertion that conversion is between byte arrays